spec/classes/profile/networking/private_spec.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+# Copyright (c) 2018 The Regents of the University of Michigan.
+# All Rights Reserved. Licensed according to the terms of the Revised
+# BSD License. See LICENSE.txt for details.
+require 'spec_helper'
+
+describe 'nebula::profile::networking::private' do
+  on_supported_os.each do |os, os_facts|
+    context "on #{os}" do
+      let(:facts) do
+        os_facts.merge(networking: { ip: '1.2.3.123' })
+      end
+
+      context 'with fully-specified parameters' do
+        let(:params) do
+          {
+            address_template: '10.0.2.%s',
+            netmask: '255.255.0.0',
+            network: '10.0.0.0',
+            broadcast: '10.0.255.255',
+            interface: 'eth1',
+          }
+        end
+
+        it do
+          is_expected.to contain_file('/etc/network/interfaces.d/private').with_content(<<~EOT)
+            auto eth1
+            iface eth1 inet static
+              address 10.0.2.123
+              netmask 255.255.0.0
+              network 10.0.0.0
+              broadcast 10.0.255.255
+          EOT
+        end
+      end
+
+      context 'with no interface' do
+        it { is_expected.not_to contain_file('/etc/network/interfaces.d/private') }
```

Comment:
This test is failing:

```  1) nebula::profile::networking::private on debian-9-x86_64 with no interface should not contain File[/etc/network/interfaces.d/private]
     Failure/Error: raise Puppet::ParseError, "has_key(): expects the first argument to be a hash, got #{args[0].inspect} which is of type #{args[0].class}"
     
     Puppet::PreformattedError:
       Evaluation Error: Error while evaluating a Method call, has_key(): expects the first argument to be a hash, got "" which is of type String (file: /home/aelkiss/Repositories/nebula/spec/fixtures/modules/nebula/manifests/profile/networking/private.pp, line: 22, column: 51) on node qoo.umdl.umich.edu
     # ./spec/fixtures/modules/stdlib/lib/puppet/parser/functions/has_key.rb:22:in `block in <module:Functions>'
     # ./spec/fixtures/modules/nebula/manifests/profile/networking/private.pp:22:in `stack'
     # :2:in `block in call_function'
     # :in `stack'
     # ./spec/classes/profile/networking/private_spec.rb:39:in `block (5 levels) in <top (required)>'
     # ------------------
     # --- Caused by: ---
     # Puppet::ParseError:
     #   has_key(): expects the first argument to be a hash, got "" which is of type String
     #   ./spec/fixtures/modules/stdlib/lib/puppet/parser/functions/has_key.rb:22:in `block in <module:Functions>'

```

Reply:
My `has_key` suggestion in private.pp should fix this.

Reply:
Still failing...